File: redo/darwin/datetimepicker.m
// 11 june 2015
#include "uipriv_darwin.h"

struct datetimepicker {
	uiDateTimePicker d;
	NSDatePicker *dp;
};

uiDefineControlType(uiDateTimePicker, uiTypeDateTimePicker, struct datetimepicker)

static uintptr_t datetimepickerHandle(uiControl *c)
{
	struct datetimepicker *d = (struct datetimepicker *) c;

	return (uintptr_t) (d->dp);
}

uiDateTimePicker *finishNewDateTimePicker(NSDatePickerElementFlags elements)
{
	struct datetimepicker *d;

	d = (struct datetimepicker *) uiNewControl(uiTypeDateTimePicker());

	d->dp = [[NSDatePicker alloc] initWithFrame:NSZeroRect];
	// TODO text field stuff
	[d->dp setDatePickerStyle:NSTextFieldAndStepperDatePickerStyle];
	[d->dp setDatePickerElements:elements];
	[d->dp setDatePickerMode:NSSingleDateMode];
	// TODO get date picker font

	uiControl(d)->Handle = datetimepickerHandle;

	return uiDateTimePicker(d);
}

uiDateTimePicker *uiNewDateTimePicker(void)
{
	return finishNewDateTimePicker(NSYearMonthDayDatePickerElementFlag | NSHourMinuteSecondDatePickerElementFlag);
}

uiDateTimePicker *uiNewDatePicker(void)
{
	return finishNewDateTimePicker(NSYearMonthDayDatePickerElementFlag);
}

uiDateTimePicker *uiNewTimePicker(void)
{
	return finishNewDateTimePicker(NSHourMinuteSecondDatePickerElementFlag);
}

File: haiku/datetimepicker.cpp
// 18 november 2015
#include "uipriv_haiku.hpp"

struct uiDateTimePicker {
	uiHaikuControl c;
	BStringView *dummy;
};

uiHaikuDefineControl(
	uiDateTimePicker,						// type name
	uiDateTimePickerType,					// type function
	dummy								// handle
)

static uiDateTimePicker *finishNewDateTimePicker(void)
{
	uiDateTimePicker *d;

	d = (uiDateTimePicker *) uiNewControl(uiDateTimePickerType());

	d->dummy = new BStringView(NULL, "TODO uiDateTimePicker not implemented");

	uiHaikuFinishNewControl(d, uiDateTimePicker);

	return d;
}

uiDateTimePicker *uiNewDateTimePicker(void)
{
	return finishNewDateTimePicker();
}

uiDateTimePicker *uiNewDatePicker(void)
{
	return finishNewDateTimePicker();
}

uiDateTimePicker *uiNewTimePicker(void)
{
	return finishNewDateTimePicker();
}
